Userland/Libraries/LibWeb/CSS/Parser/Parser.cpp

```diff
@@ -3432,6 +3432,97 @@ ErrorOr<RefPtr<CalculatedStyleValue>> Parser::parse_calculated_value(Vector<Comp
     return CalculatedStyleValue::create(calculation_tree.release_nonnull(), calc_type.release_value());
 }
 
+ErrorOr<RefPtr<StyleValue>> Parser::parse_min_function(Function const& function)
+{
+    TokenStream stream { function.values() };
+    auto parameters = parse_a_comma_separated_list_of_component_values(stream);
+
+    Vector<NonnullOwnPtr<CalculationNode>> calculated_parameters;
+    calculated_parameters.ensure_capacity(parameters.size());
+
+    CalculatedStyleValue::ResolvedType type;
+    bool first = true;
+    for (auto& parameter : parameters) {
+        auto calculation_node = TRY(parse_a_calculation(parameter));
+
+        if (first) {
+            type = calculation_node->resolved_type().value();
+            first = false;
+        }
+
+        if (calculation_node->resolved_type().value() != type) {
+            return Error::from_string_view("min() parameters must all be of the same type"sv);
+        }
+
+        calculated_parameters.append(calculation_node.release_nonnull());
+    }
+
+    NonnullOwnPtr<CalculationNode> node = TRY(MinCalculationNode::create(move(calculated_parameters)));
+    return CalculatedStyleValue::create(move(node), type);
+}
+
+ErrorOr<RefPtr<StyleValue>> Parser::parse_max_function(Function const& function)
+{
+    TokenStream stream { function.values() };
+    auto parameters = parse_a_comma_separated_list_of_component_values(stream);
```

Comment:
does this parsing guarantee we have at least one element in the parameters list?